src/mesh/parallel/fci.cxx

```diff
@@ -327,7 +327,7 @@ Field3D FCIMap::integrate(Field3D &f) const {
 void FCITransform::checkInputGrid() {
   std::string coordinates_type = "";
   if (!mesh.get(coordinates_type, "coordinates_type")) {
-    if (coordinates_type != "fci") {
+    if (strcmp(coordinates_type.c_str(), "fci") != 0) {
```

Comment:
Why this change? `coordinates_type` is a `std::string`, so comparison to a literal is fine.

Reply:
This was a strange one. Somehow between Hypnotoad and netCDF, the `coordinates_type` string has a length that is 1 longer than it should be - e.g. here coordinates_type would have length 4 - so that the comparison to a string literal fails. I think the string just has an extra null byte at the end, because any normal way of printing it only shows the actual characters. I can't see a bug in our code, neither in `tools/idllib/file_write_attribute.pro` or `src/fileio/impls/netcdf4/ncxx4.cxx` - we never set the length explicitly. So this work-around seems necessary :-(

Reply:
It does deserve a comment to explain why its doing such an ugly string comparison though...